src/group.rs

```diff
@@ -188,6 +292,20 @@ impl Encoding {
     }
 }
 
+impl TryFrom<&[u8]> for Encoding {
+    type Error = EncodingError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if bytes.len() != 32 {
+            Err(EncodingError::InvalidSliceLength)
+        } else {
```

Comment:
nit: I think this would be clearer if it were `if len == 32 { ok } else { err }` rather than `if len != 32 { err } else { ok }`, so that the `else` is the catch-all case.